### PR TITLE
Player modal

### PR DESCRIPTION
**Pull request by jbartka**

### CODE REVIEW

File: services/app/src/components/nameRoute/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import ModalDialog from 'react-bootstrap/ModalDialog';
+import ModalHeader from 'react-bootstrap/ModalHeader';
+import ModalTitle from 'react-bootstrap/ModalTitle';
+import ModalBody from 'react-bootstrap/ModalBody';
+import ModalFooter from 'react-bootstrap/ModalFooter';
```
Comment (tautvydasLiekis) [inline services/app/src/components/nameRoute/Modal.jsx:9]: Do we need this Modal.jsx file?

File: services/app/src/pages/game/Game.jsx [hunk 2]
```diff
@@ -8,19 +8,19 @@ import { useUserFromLocalStorage } from '../../hooks/useUserFromLocalStorage';
...
   useEffect(() => {
     if (username) {
-      ws.send(username);
+      ws.onopen = () => ws.send(username);
     }
   }, [username]);
 
   const handleSubmit = () => {
-    ws.send(username);
+    ws.onopen = () => ws.send(username);
```
Comment (tautvydasLiekis) [inline services/app/src/pages/game/Game.jsx:23]: Why are we adding this here?

File: services/app/src/static/style/modal.css [hunk 3]
```diff
@@ -0,0 +1,211 @@
+#title {
```
Comment (tautvydasLiekis) [inline services/app/src/static/style/modal.css:1]: Do we need this css?

File: services/app/src/components/modal/UserModal.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+};
+const UserModal = () => {
+  const [modalIsOpen, setIsOpen] = useState(false);
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
+  const submitInfo = (event) => {
+    // alert(event.target.username.value)
```
Comment (tautvydasLiekis) [inline services/app/src/components/modal/UserModal.jsx:23]: Remove this

File: services/app/package.json [hunk 5]
```diff
@@ -45,10 +45,13 @@
...
   },
   "dependencies": {
+    "bootstrap": "^5.1.3",
     "prop-types": "^15.7.2",
     "react": "^17.0.2",
+    "react-bootstrap": "^2.1.1",
     "react-copy-to-clipboard": "^5.0.4",
     "react-dom": "^17.0.2",
+    "react-modal": "^3.14.4",
     "react-router-dom": "^6.0.2"
```
Comment (tautvydasLiekis) [inline services/app/package.json:55]: No all dependencies needed, please look at other comments for reference

File: services/app/src/pages/game/Game.jsx [hunk 6]
```diff
@@ -8,19 +8,19 @@ import { useUserFromLocalStorage } from '../../hooks/useUserFromLocalStorage';
 import { WsContext } from '../../contexts/ws-context';
 
 function Room() {
-  const [roomUrl] = useState(window.location.href);
+  const [roomUrl] = useState(window.location.href.split('?')[0]);
```
Comment (tautvydasLiekis) [inline services/app/src/pages/game/Game.jsx:11]: Why is this needed?

**Comment by tautvydasLiekis** [comment]
image

Please look into these errors

File: services/app/src/pages/game/Game.jsx (hunk 2 above)
Comment (jbartka) [reply services/app/src/pages/game/Game.jsx:23]: Solves "Failed to execute 'send' on 'WebSocket': Still in CONNECTING state" problem

Comment (raywiis) [reply services/app/src/pages/game/Game.jsx:23]: I would get that in the `useEffect`, but on `handleSubmit` you'd have to click the button before the connection is open.

Maybe a better way to handle would be to show something else than the game until the connection is open?

Comment (tautvydasLiekis) [reply services/app/src/pages/game/Game.jsx]: Yes, but this was for a button logic, which is now is handled by useEffect, because after "login" modal, page is reloaded and connects to WS, from Custom Route, which checks local storage again for username

File: services/app/package.json [hunk 7]
```diff
@@ -45,10 +45,12 @@
     "webpack-dev-server": "^4.6.0"
   },
   "dependencies": {
+    "bootstrap": "^5.1.3",
```
Comment (raywiis) [inline services/app/package.json:48]: Is bootstrap necessary?

**Review by raywiis — COMMENTED** [review]
LGTM, just interested if we really need bootstrap. Personally not too into the additional `react-modal` lib, but it's not the end of the world.

File: services/app/package.json (hunk 7 above)
Comment (tautvydasLiekis) [reply services/app/package.json:48]: @jbartka is this needed?

Comment (jbartka) [reply services/app/package.json:48]: I tried removing the bootstrap from the .json file, and it seems to be working fine, so I guess we don't really need it. Besides, we are not importing this dependency into any of our directory files; thus, it would not make much sense to have it here anyways.

File: services/app/src/components/modal/UserModal.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  const [modalIsOpen, setIsOpen] = useState(false);
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
+  const submitInfo = (event) => {
+    event.preventDefault();
+    window.localStorage.setItem('user', event.target.username.value);
+    window.location.reload();
```
Comment (jvalatkevicius) [inline services/app/src/components/modal/UserModal.jsx:24]: Reloading an application on login does not seem like a right approach, as that would force user to re-download all the javascript after logging in.

I do understand, that this is needed to redirect the user to game screen, but there should be a better way. I would suggest to introduce additional route for `LoginScreen` that would contain the `UserModal` and unauthenticated users would be forced to this `LoginScreen`. Something like this should do the trick

```
const AuthenticatedRoute = (routeProps) => {
  const { authentication } = useUserFromLocalStorage();
  // set requested url as redirect URL on localStorage on context
  if (authentication) {
    return <Route {...routeProps} />;
  }

  return <Redirect to="/login" />
}

function Content() {
  return (
    <Routes>
      <Route path="/login" element={<LoginScreen />} />
      <AuthenticatedRoute path="/game/:id" element={<Game />} />
    </Routes>
  );
}
```

With this, you should be able to use from `UserModal/LoginScreen`
```
import { useNavigate } from 'react-router-dom';
<...>
const navigate = useNavigate();
const redirectUrl = // get redirect URL from localStorage or context
const submitHandler = () => {
  <...>
  navigate(redirectUrl)
}
```

File: services/app/src/components/modal/UserModal.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+  return (
+    <div>
+      <Modal
```
Comment (jvalatkevicius) [inline services/app/src/components/modal/UserModal.jsx:29]: Do we really need a Modal here? 

Looking at the code, it seems like the Modal is not really what we need here, as this modal is used on its own page, without any interactions by the user, that would open / close the modal. I would expect the modal to be used when we want to have some sort of an overlay over another content. We don't need a modal when the modal is the only component on the screen and its always open, a couple of styled `div`s would do the job just fine.

Looking at Figma designs for Landing Page, it seems like the login form does can be simply placed on its own screen as it isn't displayed over any other content in the background.

Let's get rid of this `react-modal` dependency for now, use plain HTML to implement login form and restructure the routes as I've described in the above comment.

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

**Review by jvalatkevicius — DISMISSED** [review]

**Review by jvalatkevicius — APPROVED** [review]
